etcdctl/ctlv3/command/global.go

```diff
@@ -268,3 +288,76 @@ func authCfgFromCmd(cmd *cobra.Command) *authCfg {
 
 	return &cfg
 }
+
+func insecureDiscoveryFromCmd(cmd *cobra.Command) bool {
+	discovery, err := cmd.Flags().GetBool("insecure-discovery")
+	if err != nil {
+		ExitWithError(ExitError, err)
+	}
+	return discovery
+}
+
+func discoveryCfgFromCmd(cmd *cobra.Command) *discoveryCfg {
+	var err error
+	var domainStr string
+	if domainStr, err = cmd.Flags().GetString("discovery-srv"); err != nil {
```

Comment:
simpler to treat empty string as no discovery option:
```go
domainStr, err := cmd.Flags().GetString("discovery-srv")
if err != nil {
    ExitWithError(ExitBadArgs, err)
}
return &discoveryCfg{domain: domainStr, insecure: insecureDiscoveryFromCmd(cmd)}
```

Reply:
modified